### PR TITLE
AZP/RELEASE: Publish separate jar files per platform - v1.16.x

### PR DESCRIPTION
**Pull request by Alexey-Rivkin**
## What
Publish separate JAR files per platform.

## Why ?
Publishing JAR with cross-platform support has been unsuccessful so far.

### CODE REVIEW

File: buildlib/jucx/jucx-build.yml [hunk 1]
```diff
@@ -30,10 +30,65 @@ jobs:
       - bash: |
           set -eEx
           make -s -j`nproc`
-        displayName: Build ucx
+          source buildlib/az-helpers.sh
```
Comment (artemry-nv) [inline buildlib/jucx/jucx-build.yml:33]: Please verify this shell excerpt with shellcheck and fix findings

**Review by artemry-nv — APPROVED** [review]

File: buildlib/jucx/jucx-build.yml (hunk 1 above)
Comment (Alexey-Rivkin) [reply buildlib/jucx/jucx-build.yml:33]: We had issues with the newer command substitution syntax here. Prefer to leave it as is.

File: bindings/java/src/main/native/Makefile.am [hunk 2]
```diff
@@ -9,7 +9,7 @@ jardir         = $(libdir)
 topdir         = $(abs_top_builddir)
 java_build_dir = $(builddir)/build-java
 maven_repo    ?= $(java_build_dir)/.deps
-jarfile        = $(java_build_dir)/jucx-@VERSION@.jar
+jarfile        = $(java_build_dir)/jucx-@VERSION@-$(arch).jar
```
Comment (yosefe) [inline bindings/java/src/main/native/Makefile.am:12]: do we want such suffix also for x86_64?

Comment (Alexey-Rivkin) [reply bindings/java/src/main/native/Makefile.am:12]: Don't we?

Comment (Alexey-Rivkin) [reply bindings/java/src/main/native/Makefile.am]: After consulting with stakeholders, decided to retain the original filename.
Pushed another commit.

**Review by artemry-nv — APPROVED** [review]

File: bindings/java/src/main/native/Makefile.am [hunk 3]
```diff
@@ -50,6 +50,11 @@ $(JUCX_GENERATED_H_FILES): $(STAMP_FILE)
 
 # Return CPU architecure in Java format
 arch = $(if $(filter x86_64,$(shell uname -p)),amd64,$(shell uname -p))
+
+# Only add filename suffix for ARM-based jars
+suffix = $(if $(findstring aarch64,$(shell uname -p)),-aarch64)
```
Comment (yosefe) [inline bindings/java/src/main/native/Makefile.am:55]: can we add filename suffix only if arch is NOT x86_64?

Comment (Alexey-Rivkin) [reply bindings/java/src/main/native/Makefile.am]: done

File: bindings/java/src/main/native/Makefile.am [hunk 4]
```diff
@@ -50,6 +50,11 @@ $(JUCX_GENERATED_H_FILES): $(STAMP_FILE)
 
 # Return CPU architecure in Java format
 arch = $(if $(filter x86_64,$(shell uname -p)),amd64,$(shell uname -p))
+
+# Set '-aarch64' suffix only if arch is NOT x86_64
+suffix = $(if $(filter-out x86_64,$(shell uname -p)),-aarch64)
```
Comment (yosefe) [inline bindings/java/src/main/native/Makefile.am:55]: the suffix should be the machine arch, not (necessarily) aarch64

Comment (Alexey-Rivkin) [reply bindings/java/src/main/native/Makefile.am]: done

**Review by yosefe — DISMISSED** [review]

**Comment by yosefe** [comment]
@Alexey-Rivkin can u pls squash?

**Review by yosefe — APPROVED** [review]

**Pull request merged**